src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -86,6 +83,24 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             }
         }
 
+        private void ValidateRequest()
+        {
+            if (AuthenticationRequestParameters.Scope == null || !AuthenticationRequestParameters.Scope.Any())
+            {
+                throw new MsalClientException(
+                    MsalError.ScopesRequired,
+                    MsalErrorMessage.ScopesRequired);
+            }
+
+            // Throw exception if WithAzureRegion is set to true and the cloud is not public.
+            if (_clientParameters.AutoDetectRegion && !KnownMetadataProvider.isKnownPublicEnvironment(AuthenticationRequestParameters.AuthorityInfo.Host))
```

Comment:
As discussed, let's not block the users on this.

Reply:
Removed,